chain/cosmos/chain_node.go

```diff
@@ -721,6 +722,24 @@ func (tn *ChainNode) IsAboveSDK47(ctx context.Context) bool {
 	return tn.HasCommand(ctx, "genesis")
 }
 
+// ICSVersion returns the version of interchain-security the binary was built with.
+// If it doesn't depend on interchain-security, it returns an empty string.
+func (tn *ChainNode) ICSVersion(ctx context.Context) string {
+	if strings.HasPrefix(tn.Chain.Config().Bin, "interchain-security") {
+		// This isn't super pretty, but it's the best we can do for an interchain-security binary.
+		// It doesn't depend on itself, and the version command doesn't actually output a version.
+		// Ideally if you have a binary called something like "v3.3.0-my-fix" you can use it as a version, since the v3.3.0 part is in it.
+		return semver.Canonical(tn.Image.Version)
```

Comment:
I do not quite understand this, if interchain-security as a binary is used. What if we used Gaia and/or neutron. This logic would never flow?

What about if we check the BuildDeps first. If nothing, then return `semver.Canonical(tn.Image.Version)` instead of "" ?

like
```go
func (tn *ChainNode) ICSVersion(ctx context.Context) string {
	info := tn.GetBuildInformation(ctx)
	for _, dep := range info.BuildDeps {
		if strings.HasPrefix(dep.Parent, "github.com/cosmos/interchain-security") {
			return semver.Canonical(dep.Version)
		}
	}
	return return semver.Canonical(tn.Image.Version)
}
```